### DISCUSSION TITLE
Chatbot flow

### DISCUSSION THREAD

**Discussion by harismateen** [chatwoot/chatwoot#8384] [General]
Hi,
Can I ask you guys have any plan to have advance version of chatbot flow like as attached in screenshot?
Currently, i understand that you support DialogFlow and Normal Automations Rules, 

Or is there any open source which you can suggest to have similar option as attached screenshot and can integrate with chatwoot?
![Uploading Chatbot.png…]()

**Comment by jochenstu** [comment]
With [Agent Bots](https://www.chatwoot.com/docs/product/others/agent-bots/), Chatwoot is agnostic to the bot you are building and has no limit on the complexity. As long as you return a response to Chatwoot, you can do or use whatever service you want in the middle. You can find more information behind the link above.
